hs/src/Reach/Eval/Core.hs

```diff
@@ -1013,14 +1006,23 @@ evalAsEnvM obj = case obj of
         [ ("new", retV $ public $ SLV_Prim $ SLPrim_MapCtor t) ]
   SLV_Prim SLPrim_Map ->
     Just $
-      M.fromList
-        [ ("reduce", retV $ public $ SLV_Prim $ SLPrim_MapReduce) ]
+      M.fromList $
+        [ ("reduce", retV $ public $ SLV_Prim $ SLPrim_MapReduce) ] <> foldableValueEnv
   SLV_Map _ ->
     Just $
-      M.fromList
-        [ ("reduce", delayCall SLPrim_MapReduce) ]
+      M.fromList $
+        [ ("reduce", delayCall SLPrim_MapReduce) ] <> foldableObjectEnv
+  SLV_Prim SLPrim_Set ->
```

Comment:
I think that you could just define `Set` in the standard library as `const Set = () => { const s = new Map(Null); return { ..... };` and not modify Core.hs